compass/landice/tests/ensemble_generator/plot_ensemble.py

```diff
@@ -202,61 +310,149 @@
+                               combine='nested', concat_dim='Time',
+                               decode_timedelta=False,
+                               chunks={"Time": 10})
+        yearsOutput = DS['daysSinceStart'].values[:] / 365.0
+        indices = np.nonzero(yearsOutput >= target_year)[0]
         if len(indices) > 0:
             ii = indices[0]
-            print(f'{run} using year {years[ii]}')
-            qoi_info['SLR']['values'][idx] = SLR[ii]
-
-            grdArea = f.variables['groundedIceArea'][:] / 1000.0**2  # in km^2
-            qoi_info['grd area']['values'][idx] = grdArea[ii] - grdArea[0]
 
-            iceArea = f.variables['totalIceArea'][:] / 1000.0**2  # in km^2
-            qoi_info['total area']['values'][idx] = iceArea[ii] - iceArea[0]
-
-            qoi_info['GL flux']['values'][idx] = groundingLineFlux[ii]
+            surfaceSpeed = DS['surfaceSpeed'].values[ii, :]
+            thickness = DS['thickness'].values[ii, :]
+            cellMask = DS['cellMask'].values[ii, :]
+            areaCell = DS['areaCell'].values[0, :]
+
+            # only evaluate where modeled ice exists and observed speed
+            # uncertainy is a reasonable value (<~1e6 m/yr)
+            mask = ((thickness > 0.0) *
+                    (obsSpdUnc < 0.1) *
+                    (obsSpdUnc > 0.0))
+            ind = np.nonzero(mask)[0]
+            qoi_info['speed error']['values'][idx] = \
+                ((areaCell[ind] * (surfaceSpeed[ind] - obsSpd[ind])**2 /
+                  obsSpdUnc[ind]**2).sum() / areaCell[ind].sum())**0.5
+            # float speed error
+            floatMask = (cellMask & 4) // 4
+            ind = np.nonzero(mask * floatMask)[0]
+            qoi_info['flt speed error']['values'][idx] = \
+                ((areaCell[ind] * (surfaceSpeed[ind] - obsSpd[ind])**2 /
+                  obsSpdUnc[ind]**2).sum() / areaCell[ind].sum())**0.5
+            # grd speed error
+            ind = np.nonzero(mask * np.logical_not(floatMask))[0]
+            qoi_info['grd speed error']['values'][idx] = \
+                ((areaCell[ind] * (surfaceSpeed[ind] - obsSpd[ind])**2 /
+                  obsSpdUnc[ind]**2).sum() / areaCell[ind].sum())**0.5
+        DS.close()
+
+        if plot_time_series:
+            # color lines depending on if they match obs or not
+            col = 'k'
+            alph = 0.2
+            if valid_run:
+                col = 'r'
+                alph = 0.7
 
-            qoi_info['melt flux']['values'][idx] = BMB[ii]
+            # plot time series
+            # plot 1
+            ax_ts_vaf.plot(years, VAF - VAF[0], linewidth=lw, color=col,
+                           alpha=alph)
+            if label_runs:
+                ax_ts_vaf.annotate(run, (years[-1], VAF[-1] - VAF[0]),
+                                   fontsize=6)
+            ax_ts_grvol.plot(years, grdVol - grdVol[0], linewidth=lw,
+                             color=col, alpha=alph)
+            if label_runs:
```

Comment:
Probably better to do all the plot() commands first, and then only have one `if label_runs:` with all the annotate() commands inside.

Reply:
I liked having the annotation with the corresponding chunk of code making a plot.  There are only a couple of these run labeling events so that seems not too messy.